dashboard/test/controllers/scripts_controller_test.rb

```diff
@@ -884,7 +884,10 @@ class ScriptsControllerTest < ActionController::TestCase
     Rails.application.config.stubs(:levelbuilder_mode).returns true
 
     unit = create :script
-    stub_file_writes(unit.name)
+    filenames_to_stub = ["#{Rails.root}/config/scripts/#{unit.name}.script", "#{Rails.root}/config/scripts_json/#{unit.name}.script_json", "#{Rails.root}/config/course_offerings/fake-family-z.json"]
+    File.stubs(:write).with do |filename, _|
+      filenames_to_stub.include?(filename)
+    end
```

Comment:
can you help me understand why only this test needs to stub a different set of files, but the rest of the tests which call stub_file_writes do not?

Reply:
This tests specifically kept failing with the `|| filename.end_with?('scripts.en.yml')`. I spent a bunch of time trying to figure out what was going on and didn't get the exact reason locked down. I can take another look. Interested if you have any ideas why

Reply:
the error is:
```
        Response body: {"base":["undefined method `end_with?' for #\u003cPathname:0x00007fae706a7040\u003e"]}
```
the reason is that File.write can be called with either a string or a Pathname. The solution is to add `to_s` like this:
```
filename.to_s.end_with?('scripts.en.yml')
```

Reply:
it should be fine to make that change within `stub_file_writes`.

Reply:
ahhhh thank you!